File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/MimicSummoner.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.entity.level.entities.chest.Chest;
import org.rexcellentgames.burningknight.entity.level.entities.chest.Mimic;

public class MimicSummoner extends Equipable {
	{
		sprite = "item-mimic_summoner";
	}

	@Override
	public void onEquip() {
		super.onEquip();
		Mimic.chance += 100;

		for (Chest chest : Chest.all) {
			chest.toMimic();
		}
	}

	@Override
	public void onUnequip() {
		super.onUnequip();
		Mimic.chance -= 100;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/MimicSummoner.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.entity.level.entities.chest.Chest
import org.rexcellentgames.burningknight.entity.level.entities.chest.Mimic

class MimicSummoner : Equippable() {
	init {
		sprite = "item-mimic_summoner"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		Mimic.chance += 100f

		for (chest in Chest.all) {
			chest.toMimic()
		}
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		Mimic.chance -= 100f
	}
}